client/src/root.js

```diff
@@ -35,7 +45,117 @@ if (mailtrainConfig.reportsEnabled) {
 
 
 @withComponentMixins([
-    withTranslation
+    withTranslation,
+    withForm,
+    withErrorHandling,
+    withPageHelpers,
+    requiresAuthenticatedUser
+])
+class PreviewNamespaceFilterModalDialog extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {namespaceId: null, namespaceName: this.props.t("namespaceFilter")};
+        this.initForm({
+            leaveConfirmation: false
+        });
+    }
+
+    static propTypes = {
+        visible: PropTypes.bool.isRequired,
+        onHide: PropTypes.func.isRequired,
+    }
+
+    localValidateFormValues(state) {
+        const t = this.props.t;
+        if (!state.getIn(['namespace', 'value']) && state.getIn(['namespace_filter_enabled', 'value'])) {
```

Comment:
For fields that are not directly reflected in DB, I use camel case. I.e. namespaceFilterEnabled.